pkg/controller/virtualmachinevolumeexport/virtualmachinevolumeexport_controller.go

```diff
@@ -95,34 +95,41 @@ func (r *ReconcileVirtualMachineVolumeExport) Reconcile(request reconcile.Reques
 	}
 	r.vmvExport = cachedVmvExport.DeepCopy()
 
-	syncAll := func() error {
-		// check if virtual machine volume to export is available
-		if err := r.validateVirtualMachineVolume(); err != nil {
-			return err
-		}
-		// if there is no pvc, update state to creating and create pvc
-		if err := r.syncExportPvc(); err != nil {
-			return err
-		}
-		// if pvc export is not completed, create exporter pod if it not exist
-		// if there is an exporter pod and state is complete, update completed to yes and delete it
-		if err := r.syncExporterPod(); err != nil {
-			return err
+	// check if virtual machine volume to export is available
+	if err := r.validateVirtualMachineVolume(); err != nil {
+		if err2 := r.updateStateWithReadyToUse(hc.VirtualMachineVolumeExportStatePending, corev1.ConditionFalse, "VmvExportIsInPending", err.Error()); err2 != nil {
+			return reconcile.Result{}, err2
 		}
-		// if destination is local, create local pod and update readytouse to true if it not exist
-		if destination := r.getDestination(); destination == ExporterDestinationLocal {
-			if err := r.syncLocalPod(); err != nil {
-				return err
-			}
+		return reconcile.Result{RequeueAfter: hc.VirtualMachineVolumeExportReconcileAgain}, nil
+	}
+
+	// if there is no pvc, update state to creating and create pvc
+	if err := r.syncExportPvc(); err != nil {
```

Comment:
How abount merge following 3 cases?  `syncExportPvc, syncExporterPod, syncLocalPod`

Reply:
Code will be simpler. Thanks!